code/game/turfs/simulated/wall_rot.dm

```diff
@@ -0,0 +1,33 @@
+/obj/effect/overlay/wallrot
+	name = "wallrot"
+	desc = "Ick..."
+	icon = 'icons/effects/wallrot.dmi'
+	icon_state = "wallrot"
+	anchored = TRUE
+	density = TRUE
+	layer = 5
+	mouse_opacity = 0
+
+/obj/effect/overlay/wallrot/Initialize(mapload, ...)
+	. = ..()
+	pixel_x += rand(-10, 10)
+	pixel_y += rand(-10, 10)
+
+/obj/effect/overlay/wallrot/proc/scrape(var/mob/user)
+	var/obj/item/rot_sample/RS = new /obj/item/rot_sample(get_turf(src))
+	user.put_in_hands(RS)
+	qdel(src)
+
+/obj/item/rot_sample
+	name = "rot sample"
+	desc = "A gross, wet, squishy piece of what may be a plant."
+	desc_info = "This sample can be ground to retrieve reagents inside it."
+	icon = 'icons/effects/wallrot.dmi'
+	icon_state = "rot_sample"
+
+/obj/item/rot_sample/Initialize(mapload)
+	. = ..()
+	create_reagents(15)
+	var/list/possible_reagents = list(/datum/reagent/bicaridine, /datum/reagent/kelotane, /datum/reagent/dexalin, /datum/reagent/mortaphenyl, /datum/reagent/oculine, /datum/reagent/peridaxon)
```

Comment:
Would it be stupid (or hard to implement) picking a single reagent type (or maybe two/three) per "wall rot start" and keep the same composition throughout the whole growth? Would make more sense IMO.

Reply:
i don't think it'd be difficult per se, but i think it'd reduce the value of the event, since during one event, you'd have a good chance of getting a lot of different good chems